third_party/amd/lib/TritonAMDGPUTransforms/RescheduleOps.cpp

```diff
@@ -66,11 +79,60 @@ struct Node {
 
 private:
   Operation *op;
+  int32_t priority;
   llvm::SetVector<Node *> realChildren;
   llvm::SetVector<Node *> artificialChildren;
   llvm::SetVector<Node *> parents;
 };
 
+struct PriorityStrategy {
```

Comment:
I'm not sure where this "PriorityStrategy" fits in to our overall architecture which we've been putting together in the iSched document. Please create a comment here explaining the design of this Priority code, and explain how it relates to rest of iSched. I'm not quite sure if this is the critical path analysis we were discussing earlier or something different. We just need a high-level design to help us grok the code.